<launch>

  <arg name="tello_driver_name" default="tello_driver_node" />

  <!-- Flags -->
  <arg name="flag_use_vel_cmd_unstamped" default="False" />

  <!-- Topics -->
  <arg name="tello_takeoff_topic_name" default="/tello/takeoff" />
  <arg name="tello_land_topic_name" default="/tello/land" />
  <arg name="tello_image_topic_name" default="/tello/camera/image_raw" />
  <arg name="tello_flight_data_topic_name" default="/tello/flight_data" />
  <arg name="tello_vel_cmd_unstamped_topic_name" default="/tello/cmd_vel_unstamped" />
  <arg name="tello_vel_cmd_stamped_topic_name" default="/tello/cmd_vel_stamped" />
  <arg name="tello_collision_topic_name" default="/tello/is_colliding" />

  <!-- Tello WiFi credentials-->
  <arg name="tello_ssid" default="tello" />
  <arg name="tello_password" default="" />
  <arg name="connect_to_tello_wife_auto" default="True" />

  <!-- TF2 Frames -->
  <arg name="tello_frame_name" default="tello_base_link" />
  <arg name="tello_world_frame_name" default="world" />

  <node name="$(arg tello_driver_name)" pkg="tello_driver" type="tello_driver_ros_node.py">
  
    <param name="tello_takeoff_topic_name" value="$(arg tello_takeoff_topic_name)" type="string" />
    <param name="tello_land_topic_name" value="$(arg tello_land_topic_name)" type="string" />
    <param name="tello_image_topic_name" value="$(arg tello_image_topic_name)" type="string" />
    <param name="tello_flight_data_topic_name" value="$(arg tello_flight_data_topic_name)" type="string" />
    <param name="tello_vel_cmd_unstamped_topic_name" value="$(arg tello_vel_cmd_unstamped_topic_name)" type="string" />
    <param name="tello_vel_cmd_stamped_topic_name" value="$(arg tello_vel_cmd_stamped_topic_name)" type="string" />
    <param name="tello_collision_topic_name" value="$(arg tello_collision_topic_name)" type="boolean" />

    <param name="tello_frame_name" value="$(arg tello_frame_name)" type="string" />
    <param name="tello_world_frame_name" value="$(arg tello_world_frame_name)" type="string" />

    <param name="flag_sub_tello_vel_cmd_unstamped" value="$(arg flag_use_vel_cmd_unstamped)" type="boolean" />

    <param name="tello_ssid" value="$(arg tello_ssid)" type="string" />
    <param name="tello_pw" value="$(arg tello_password)" type="string" />
    <param name="connect_to_tello_wife_auto" value="$(arg connect_to_tello_wife_auto)" type="boolean" />

  </node>

</launch>
